src/lib/proposer/proposer.ml

```diff
@@ -472,16 +480,17 @@ let run ~logger ~prover ~verifier ~trust_system ~get_completed_work
                 | `Propose_now (keypair, data) ->
                     Interruptible.finally
                       (Singleton_supervisor.dispatch proposal_supervisor
-                         (keypair, data))
+                         (keypair, now, data))
```

Comment:
`next_proposal` isn't necessarily cheap. Especially on low stake accounts. But is it slow enough that we can't use `now`? I'm not sure. Should this be the `scheduled_time`?

Reply:
According to [the VRF benchmark](https://github.com/CodaProtocol/coda/pull/2773) it's about ~1ms/eval, with `k=2160` that leaves 51840ms ~ 52s to calculate that an entire epoch was lost. 

.. But if `next_proposal` returns `Propose_now` of course we won early enough that it's _now_ and we don't spend time on slots after now. So I guess it shouldn't matter.